pkg/component/spec.go

```diff
@@ -81,9 +81,22 @@ func (t *CommandTimeoutSpec) InitDefaults() {
 	t.Stop = 30 * time.Second
 }
 
+// ServiceTimeoutSpec is the timeout specification for subprocess.
+type ServiceTimeoutSpec struct {
+	Checkin time.Duration `config:"checkin" yaml:"checkin"`
```

Comment:
Seems you added the timeout to the specification, but don't use it in the service runtime.

Reply:
+1

Reply:
it's coming, in the other chunk, wip